### PR TITLE
[chore] Require all checks to pass on a PR

### PR DESCRIPTION
**Pull request by danielsn**
# What does this PR do?

Adds a CI job to ensure that all checks pass on a PR.

# Motivation

Currently, marking checks "required" is manual, and hard to do for lots of tests.  This automates that.

# Additional Notes

https://github.com/marketplace/actions/allcheckspassed

# How to test the change?

Describe here in detail how the change can be validated.

### CODE REVIEW

File: .github/workflows/all-checks.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    allchecks:
+        runs-on: ubuntu-latest
+        permissions:
+            checks: read
+            contents: read
+        steps:
+          - uses: wechuli/allcheckspassed@v1
+            with:
+                delay: '3'
+                retries: '30'
```
Comment (morrisonlevi) [inline .github/workflows/all-checks.yml:17]: 30 retries?

Comment (danielsn) [reply .github/workflows/all-checks.yml:17]: It polls once a minute to see if the PR has succeeded, up to `retries` times.  Our seems to take a bit less than 20 minutes: 30 seemed like a reasonable buffer

File: .github/workflows/all-checks.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+name: All checks pass
```
Comment (bantonsson) [inline .github/workflows/all-checks.yml:1]: Could this be named `Required`, since we might have checks that we would like to exclude at some point.

**Review by bantonsson — APPROVED** [review]
Very nice. Only minor comment about workflow name.

**Pull request merged**